netconf_client/ncclient.py

```diff
@@ -21,40 +24,258 @@
 )
 
 
+# Defines the scope for netconf traces
+logger = logging.getLogger("netconf_client_pretty")
+
+
+def pretty_xml(xml):
+    """Reformats a given string containing an XML document (for human readable output)"""
+
+    parser = etree.XMLParser(remove_blank_text=True)
+    tree = etree.fromstring(xml, parser)
+    return etree.tostring(tree, pretty_print=True).decode()
+
+
 class Manager:
-    """A helper class for performing common NETCONF operations
+    """A helper class for performing common NETCONF operations with pretty logging.
 
     This class attempts to be API compatible with the manager object
     from the original ncclient.
 
     This class is also a context manager and can be used with `with`
     statements to automatically close the underlying session.
 
+    NetConf requests and responses are logged using the 'netconf_client' scope.
+    The API functions, which are to be logged, can be configured (default: all but 'close_session').
+    Also the log level can be changed via API (logger.DEBUG is the default).
+
+    Each log entry shows a log ID (the peer's IP address as default).
+    Also a timestamp is printed for each request/response message and the round-trip delay
```

Comment:
The Python logging system already does timestamping.

Reply:
You are right. I can remove the timestamps from the output, provided there is a way to enable time stamps from klish. I have no idea how to do this.
On the other hand, for measuring the round-trip delay I still need some kind of timestamps (I could even use the stead clock here). The delay seems to be very helpful and should not be removed.

Reply:
Agreed, the delay is useful info

Reply:
I leave it at the real time clock because monotonic timestames were only introduced with Python 3.3 (and netconf_client should be compatible with Python 2.x as far as I know).

Reply:
Correct, `netconf_client` is remaining 2.7 compatible for a little while longer. Though that is something to revisit soon most likely.